cpp/util/etcd.cc

```diff
@@ -746,17 +784,28 @@ void EtcdClient::FetchDone(RequestState* etcd_req, Task* task) {
 
 EtcdClient::HostPortPair EtcdClient::GetEndpoint() const {
   lock_guard<mutex> lock(lock_);
-  return endpoint_;
+  return etcds_.front();
 }
 
 
 EtcdClient::HostPortPair EtcdClient::UpdateEndpoint(const string& host,
                                                     uint16_t port) {
   lock_guard<mutex> lock(lock_);
-  VLOG_IF(1, endpoint_.first != host || endpoint_.second != port)
-      << "new endpoint: " << host << ":" << port;
-  endpoint_ = make_pair(host, port);
-  return endpoint_;
+  HostPortPair new_endpoint(host, port);
+  auto it(find(etcds_.begin(), etcds_.end(), new_endpoint));
+  if (it == etcds_.end()) {
+    // TODO(alcutter): We don't really have a way of knowing when to remove
+    // etcd endpoints at the moment.  We should really be querying the etcd
+    // cluster for its members and using that list.
+    etcds_.emplace_front(move(new_endpoint));
+  } else {
+    etcds_.emplace_front(move(*it));
+    etcds_.erase(it);
```

Comment:
I think you could use `std::list::splice()` to do this, but this is all right too.